wgpu-core/src/device/mod.rs

```diff
@@ -2735,10 +2748,13 @@ impl<A: HalApi> Device<A> {
         let format_desc = format.describe();
         self.require_features(format_desc.required_features)?;
 
-        if self
+        let using_device_features = self
             .features
-            .contains(wgt::Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES)
-        {
+            .contains(wgt::Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES);
+        // If we're running downlevel, we need to manually ask the backend what we can use as we can't trust WebGPU.
```

Comment:
That's a bit concerning. If we can't respect WebGPU guarantees on texture formats, then we'll need to set downlevel baseline for them. This code implies that the downlevel baseline is basically zero capabilities, and it would make peoples lives really miserable

Reply:
I agree, though I need to more fully establish the exact features allowed by various backends before I start to build that out. Let's leave this as is and I'll address it in a follow up -- I'll file an issue.